applications/IgaApplication/custom_conditions/sbm_laplacian_condition.cpp

```diff
@@ -0,0 +1,550 @@
+            for (IndexType idim = 0; idim < dim; idim++) {
+                    DN_dot_n(0, i)   += DN_DX(i, idim) * normal_physical_space[idim];
+                    DN_dot_n_vec(i)  += DN_DX(i, idim) * normal_physical_space[idim];         
+            } 
+            // Reset for each node
+            double H_taylor_term = 0.0; 
+
+            if (dim == 2) {
+                for (IndexType n = 1; n <= basis_functions_order; n++) {
+                    // Retrieve the appropriate derivative for the term
+                    Matrix& shapeFunctionDerivatives = mShapeFunctionDerivatives[n-1];
+                    for (IndexType k = 0; k <= n; k++) {
+                        IndexType n_k = n - k;
+                        double derivative = shapeFunctionDerivatives(i,k); 
+                        // Compute the Taylor term for this derivative
+                        H_taylor_term += computeTaylorTerm(derivative, d[0], n_k, d[1], k);
+                    }
+                }
+            } else {
+                // 3D
+                for (IndexType n = 1; n <= basis_functions_order; n++) {
+                    Matrix& shapeFunctionDerivatives = mShapeFunctionDerivatives[n-1];
+                    
+                    int countDerivativeId = 0;
+                    // Loop over blocks of derivatives in x
+                    for (IndexType k_x = n; k_x >= 0; k_x--) {
+                        // Loop over the possible derivatives in y
+                        for (IndexType k_y = n - k_x; k_y >= 0; k_y--) {
+                            
+                            // derivatives in z
+                            IndexType k_z = n - k_x - k_y;
+                            double derivative = shapeFunctionDerivatives(i,countDerivativeId); 
+
+                            H_taylor_term += computeTaylorTerm3D(derivative, d[0], k_x, d[1], k_y, d[2], k_z);
+                            countDerivativeId++;
+                        }
+                    }
+                }
+            }
+            
+            H_sum(0,i) = H_taylor_term + H(0,i);
+            H_sum_vec(i) = H_taylor_term + H(0,i);
+        }
+
+        // Assembly
+        // -(GRAD_w * n, u + GRAD_u * d + ...)
+        noalias(rLeftHandSideMatrix) -= Guglielmo_innovation * prod(trans(DN_dot_n), H_sum)  * r_integration_points[0].Weight() ; // * std::abs(DetJ0) ;
+        // -(w,GRAD_u * n) from integration by parts -> Fundamental !! 
+        noalias(rLeftHandSideMatrix) -= prod(trans(H), DN_dot_n)                             * r_integration_points[0].Weight() ; // * std::abs(DetJ0) ;
+        // SBM terms (Taylor Expansion) + alpha * (w + GRAD_w * d + ..., u + GRAD_u * d + ...)
+        noalias(rLeftHandSideMatrix) += prod(trans(H_sum), H_sum) * penalty_integration ;
+
+
+        if (CalculateResidualVectorFlag) {
+            
+            const double u_D_scalar = candidateClosestSkinSegment1.GetGeometry()[closestNodeId].GetValue(TEMPERATURE);

```

Comment:
You're assuming `TEMPERATURE` as unknown variable. This must be taken from the convection diffusion settings.